tests/mobilebert/test_tokenization_mobilebert.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+# Copyright 2022 Leon Derczynski. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the MobileBERT tokenizer. """
+
+
+import filecmp
+import unittest
+from tempfile import NamedTemporaryFile
+
+import requests
+from transformers import MobileBertTokenizer, MobileBertTokenizerFast
+from transformers.testing_utils import require_tokenizers
+
+from ..bert.test_tokenization_bert import BertTokenizationTest
+
+
+# disable duplicate incorporation of tests from parent class in this module
+BertTokenizationTest.__test__ = False
+
+
+@require_tokenizers
+class MobileBertTokenizationTest(BertTokenizationTest, unittest.TestCase):
```

Comment:
To facilitate future maintenance of this test, I think it would be easier to just copy and paste the Bert's test into this file so that `MobileBertTokenizationTest` does not depend on `BertTokenizationTest`. In transformers - in general - we prefer to copy/paste code rather than create inheritances between models :slightly_smiling_face:

Reply:
Ah, OK. In this case the solution is trivial of course. Might I ask what the reasoning is behind the preference to copy code instead of minimise duplication, just so I know in future?

Reply:
I think [this part of our documentation ](https://huggingface.co/docs/transformers/add_new_model#code-style )should interest you! 

I think for the particular case of MobileBert it is not obvious because I think that today we would not have created a `MobileBertTokenizer` class but only referenced the `BertTokenizer` in the config of the checkpoints if we really don't want to have any changes between the 2 tokenizers.

Now that the `MobileBertTokenizer` class exists I think the spirit is rather that `MobileBertTokenizer`'s behaviour should not be constrained by the BERT model. 

I would like to ping @LysandreJik - main maintainer of `transformers` - on this subject if he has a different opinion than me on this issue, especially because it's a design discussion :slightly_smiling_face: .

Reply:
@patrickvonplaten has a recent blog post for this 😄 

https://huggingface.co/blog/transformers-design-philosophy

Reply:
Perfect, thanks! I especially like points 3+4.

Re: tests - I wonder what we do if a bug is found for the Bert tokenizer, in order to get the resulting test also implemented in eg. MobileBERT's tests. Where is that provenance annotated? Just in the ancestors? That said, I suppose new BERT tokenizer bugs are unlikely at this point :)

Reply:
There are things like (only for models/tokenizers, etc. for now, not for tests)
```
# Copied from <predecessor_model>.<function>
```

For example,

https://github.com/huggingface/transformers/blob/21decb7731e998d3d208ec33e5b249b0a84c0a02/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L325

Reply:
For tests, I think we haven't really focus on this point (yet)